modules/programs/vdirsyncer.nix

```diff
@@ -0,0 +1,269 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vdirsyncer;
+
+  vdirsyncerCalendarAccounts =
+    filterAttrs (_: v: v.vdirsyncer.enable)
+    (mapAttrs' (n: v: nameValuePair ("calendar_" + n) v)  config.accounts.calendar.accounts);
+
+  vdirsyncerContactAccounts =
+    filterAttrs (_: v: v.vdirsyncer.enable)
+    (mapAttrs' (n: v: nameValuePair ("contacts_" + n) v)  config.accounts.contact.accounts);
+
+  vdirsyncerAccounts = vdirsyncerCalendarAccounts // vdirsyncerContactAccounts;
+
+  wrap = s: ''"${s}"'';
+
+  listString = l: ''[${concatStringsSep ", " l}]'';
+
+  boolString = b: if b then "true" else "false";
+
+  localStorage = a:
+  filterAttrs (_: v: v != null)
+  ((getAttrs [ "type" "fileExt" "encoding" ] a.local) // {
+    path = a.local.path;
+    postHook = if a.vdirsyncer == null
+      then null
+      else a.vdirsyncer.postHook;
```

Comment:
I'd suggest writing the post hook commands to a shell script, e.g. to have something like
```
postHook =
  pkgs.writeShellScriptBin "post-hook" a.vdirsyncer.postHook
  + "/bin/post-hook";
```
This will allow the use of multiline scripts and the `types.lines` type.